execution/src/windows/execution.cc

```diff
@@ -117,11 +118,11 @@ namespace leatherman { namespace execution {
 
         // Format a name for the pipe based on the process and counter
         wstring name = boost::nowide::widen((boost::format("\\\\.\\Pipe\\leatherman.%1%.%2%") %
-            GetCurrentProcessId() %
+            GetCurrentThreadId() %
```

Comment:
`CreateNamedPipe` will create a pipe that is visible to other processes, which is why they're commonly used for IPC -- the named pipe client connects to a pipe using the same name that the named pipe server used when creating the pipe. So I think we need to include both process and thread id when constructing the name. Also are there are any security issues with allowing other processes to connect to the named pipe?

Can we instead create an anonymous pipe using [`CreatePipe`](https://msdn.microsoft.com/en-us/library/windows/desktop/aa365152(v=vs.85).aspx) avoiding the issue altogether?